packages/@react-spectrum/list/src/listview.css

```diff
@@ -87,12 +98,15 @@
     padding-top: 0px;
     padding-bottom: 0px;
   }
-}
-
-.react-spectrum-ListViewItem-selected {
-  background-color: var(--spectrum-table-row-background-color-selected, var(--spectrum-alias-highlight-selected));
-  &.is-hovered {
-    background-color: var(--spectrum-table-quiet-row-background-color-selected-hover);
+  
+  /* give first and last items border-radius to match listview container */
+  div:first-child > div[role="row"] > & {
+    border-start-start-radius: var(--spectrum-listview-item-start-end-border-radius);
+    border-start-end-radius: var(--spectrum-listview-item-start-end-border-radius);
+  }
+  div:last-child > div[role="row"] > & {
+    border-end-start-radius: var(--spectrum-listview-item-start-end-border-radius);
+    border-end-end-radius: var(--spectrum-listview-item-start-end-border-radius);
```

Comment:
I think this is what we want. Gives the first and last items a border radius that is the ListView's border radius minus the ListView's border width:

<img width="217" alt="Screen Shot 2021-11-05 at 2 18 14 PM" src="https://user-images.githubusercontent.com/8961049/140573301-5642800d-e1ea-4fd3-a9ca-768d3cd7ce92.png">

And this selector is a bit awkward so if anyone can recommend a better one, let me know.

Reply:
selector seems fine, it's traverses right to left, so the list item is already pretty specific, shouldn't cause any performance issues
i didn't realize that the items literally swapped every time one scrolls out of view, so this rounded border is technically always applied to the last element in the view, which is great, even while scrolling